week3/runoff/runoff.c

```diff
@@ -0,0 +1,201 @@
+    {
+        printf("Maximum number of candidates is %i\n", MAX_CANDIDATES);
+        return 2;
+    }
+    for (int i = 0; i < candidate_count; i++)
+    {
+        candidates[i].name = argv[i + 1];
+        candidates[i].votes = 0;
+        candidates[i].eliminated = false;
+    }
+
+    voter_count = get_int("Number of voters: ");
+    if (voter_count > MAX_VOTERS)
+    {
+        printf("Maximum number of voters is %i\n", MAX_VOTERS);
+        return 3;
+    }
+
+    for (int i = 0; i < voter_count; i++)
+    {
+
+        for (int j = 0; j < candidate_count; j++)
+        {
+            string name = get_string("Rank %i: ", j + 1);
+
+            if (!vote(i, j, name))
+            {
+                printf("Invalid vote.\n");
+                return 4;
+            }
+        }
+
+        printf("\n");
+    }
+
+    while (true)
+    {
+        tabulate();
+
+        bool won = print_winner();
+        if (won)
+        {
+            break;
+        }
+
+        int min = find_min();
+        bool tie = is_tie(min);
+
+        if (tie)
+        {
+            for (int i = 0; i < candidate_count; i++)
+            {
+                if (!candidates[i].eliminated)
+                {
+                    printf("%s\n", candidates[i].name);
+                }
+            }
+            break;
+        }
+
+        eliminate(min);
+
+        for (int i = 0; i < candidate_count; i++)
+        {
+            candidates[i].votes = 0;
+        }
+    }
+    return 0;
+}
+
+bool vote(int voter, int rank, string name)
+{
+    for (int i = 0; i < candidate_count; i++)
+    {
+        if (strcmp(candidates[i].name, name) == 0)
+        {
+            preferences[voter][rank] = i;
+            return true;
+        }     
+    }
+    
+        return false;
+}
+
+void tabulate(void)
+{
+    for (int i = 0; i < voter_count; i++)
+    {
+        for (int j = 0; j < candidate_count; j++)
+        {
+            if (candidates[preferences[i][j]].eliminated == false)
+            {
+                candidates[preferences[i][j]].votes++;
+                break;
+            }
+        }
+    }
+    
+    return;
+}
+
+bool print_winner(void)
+{
+    for (int i = 0; i < candidate_count; i++)
+    {
+        bool candidateHasAbsoluteMajorityOfVotes = candidates[i].votes > voter_count/2;
+        if (candidateHasAbsoluteMajorityOfVotes)
+        {
+            printf("%s\n", candidates[i].name);
+            return true;
+        }     
+    }
+    
+    return false;
+}
+
+int find_min(void)
+{
+    int min = voter_count;
+    for (int i = 0; i < candidate_count; i++)
+    {
+        if (candidates[i].votes < min && candidates[i].eliminated == false) 
+        {
+            min = candidates[i].votes;
+        }
+        
+    }
+    
+    return min;
+}
+
+bool is_tie(int min)
+{
+    int eliminate = 0;
+    int counter = 0;
+    for (int i = 0; i < candidate_count; i++)
+    {
+        if (!candidates[i].eliminated)
```

Comment:
```suggestion
        if (!candidates[i].eliminated)
```

Sometimes you use `!eliminated` and other times `eliminated == false`. Choose one